File: public/docs/_examples/forms/dart/angular2_forms-4-DIAGNOSTIC/lib/hero.dart
// #docregion
library hero_form.hero;

class Hero {
  int number;
  String name;
  String power;
  String alterEgo;

  Hero(this.number, this.name, this.power, [this.alterEgo]);

  String toString() => '$number: $name ($alterEgo). Super power: $power';
}

File: public/docs/_examples/testing/ts/app/hero.ts
// #docregion
let nextId = 30;

export class Hero {
	constructor(
		public id?: number,
		public name?: string,
		public power?: string,
		public alterEgo?: string
	) {
		this.id = id || nextId++;
	}

	clone() { return Hero.clone(this); }

	static clone = (h:any) => new Hero(h.id, h.name, h.alterEgo, h.power);

	static setNextId(next:number) { nextId = next; }
}
